### ISSUE TITLE
Cannot reflect function with empty pattern match

### ISSUE THREAD

**Issue by edsko** [ucsd-progsys/liquidhaskell#1161]
This is related to #890, but not the same problem:

``` haskell
{-@ data Void @-}
data Void

{-@ reflect absurd @-}
absurd :: Void -> a
absurd v = case v of {}
```

yields

```
 <no location info>: Error: Uh oh.
     (..): Error: Cannot lift Haskell function `absurd` to logic
    Bah
```

I can work around it using

``` haskell
{-@ reflect bottom @-}
{-@ lazy bottom @-}
bottom :: a
bottom = bottom

{-@ data Void @-}
data Void

{-@ reflect absurd @-}
absurd :: Void -> a
absurd v = bottom
```

(though not using `error` because that is not reflected).

**Comment by nikivazou** [comment]
That's a smart work-around!!!

Reflection of diverging and absurd in the logic sound scary/challenging for the metatheory, but it is not entirely clear if we should allow it or not. 

BTW, I am super interested on what you are trying to prove! Feel free to pm me and let me know.

**Comment by edsko** [comment]
I agree that reflection of `bottom` is a dubious matter, but reflection of `absurd` is perfectly sound surely? It's the Curry-Howard reflection of reductio ad absurdum -- and since the right hand side will therefore never actually matter, I figured using `bottom` there is okay. The only concern would be if the very presence of a rule like `bottom = bottom` might cause LH or Z3 to loop or (worse) prove incorrect things, but from my (very limited!) understanding of how Liquid Haskell works I didn't that that would be the case? 

Will pm you :)
